app/ai-gateway/streaming.md

```diff
@@ -0,0 +1,149 @@
+---
+title: "Streaming with AI Gateway"
+content_type: reference
+layout: reference
+
+products:
+  - gateway
+
+tags:
+  - ai-gateway
+  - streaming
+
+plugins:
+  - ai-proxy
+  - ai-proxy-advanced
+
+min_version:
+  gateway: '3.7'
+
+description: This guide walks you through setting up the AI Proxy and AI Proxy Advanced plugin with streaming.
+---
+
+## What is request streaming
+
+In an LLM (Large Language Model) inference request, {{site.base_gateway}} uses the upstream provider's REST API to generate the next chat message from the caller. 
+Normally, this request is processed and completely buffered by the LLM before being sent back to {{site.base_gateway}} and then to the caller in a single large JSON block. This process can be time-consuming, depending on the `max_tokens`, other request parameters, and the complexity of the request sent to the LLM model.
+
+To avoid making the user wait for their chat response with a loading animation, most models can stream each word (or sets of words and tokens) back to the client. This allows the chat response to be rendered in real time.
+	
+For example, a client could set up their streaming request using the OpenAI Python SDK like this:
+
+```python
+from openai import OpenAI
+
+client = OpenAI(
+    base_url="http://127.0.0.1:8000/12/openai",
+    api_key="none"
+)
+
+stream = client.chat.completions.create(
+    model="gpt-4",
+    messages=[{"role": "user", "content": "Tell me the history of Kong Inc."}],
+    stream=True,
+)
+
+print('>')
+for chunk in stream:
+    print(chunk.choices[0].delta.content or "", end="", flush=True)
+```
+
+The client won't have to wait for the entire response. Instead, tokens will appear as they come in, for example:
+
+```sh
+$ python3 long-streaming-request.py
+> Kong Inc. is a software company providing cloud-native connectivity solutions for APIs and....
```

Comment:
Let's split the response out into a separate codeblock.

Reply:
I've removed this example, I don't think it's really helpful